### PR TITLE
fix zson escape parsing and handling of unset

### PR DESCRIPTION
**Pull request by mccanne**
These changes add the TBD code for unescaping escaped characters
in zson input.  There is also a check for a single "-" for a value
field, which now correctly turns into an unset column.

### CODE REVIEW

**Comment by mccanne** [comment]
Yes, this needs tests.

File: pkg/zson/raw.go [hunk 1]
```diff
@@ -203,19 +249,32 @@ func zsonParseField(b []byte) ([]byte, []byte, error) {
...
+			if to == 1 && b[0] == '-' {
+				return nil, b[2:], nil
+			}
+			return b[:to], b[from+1:], nil
 		case backslash:
 			// XXX need to implement full escape parsing,
 			// for now just skip one character
-			i += 1
+			from += 1
+			e, n := unescape(b[from:])
```
Comment (nwt) [inline pkg/zson/raw.go]: I think we should revise the ZSON spec so that `\\` and `\xdd` are the only escape sequences and then use pkg/zeek.Unescape here.

Comment (mccanne) [reply pkg/zson/raw.go]: +1

Comment (mccanne) [reply pkg/zson/raw.go]: zeek.Unescape allocates new memory whereas this code unescapes in place

**Comment by mccanne** [comment]
I took the liberty of folding these changes into #42

**Pull request closed without merging**